### ISSUE TITLE
FSAL Ceph Attr_Expiration_Time multiple MDS corrupt metadata

### ISSUE THREAD

**Issue by lebonez** [nfs-ganesha/nfs-ganesha#563]
OS: Centos 7.7.1908
Versions:
Ceph - 14.2.8
NFS-Ganesha - 2.8.3

My setup is a ceph cluster with a single filesystem that is shared by ganesha and a couple kernel mounts. The export configs are in rados. The ceph config is default with everything in regards to cephfs/mds. I do not see anything in logs on ganesha side except one error at the bottom with debug on. Ceph side gave a few errors messages which I will list in the ISSUE portion.

TLDR;
Multiple MDS with the Attr_Expiration_Time set in EXPORT_DEFAULTS would cause 'Backtrace' corrupt metadata. The MDSes were also saying ganesha client's were not responding to client pressure. All operations would hang for the directory with MDS reports slow requests during high traffic. Scrub repairing metadata, then going back to 3 MDSes didn't help same issue came right back once traffic resumed. Dropping back down to a single MDS and scrubbing showed same 'Backtrace' corrupt metadata. I stayed with 1 MDS and didn't have any issues with the Attr_Expiration_Time set. The ganesha clients were still failing to respond to cache pressure. Finally I removed Attr_Expiration_Time parameter and went back to 3 MDSes and didn't have any issues. Any ideas? 

LONG ISSUE:
I ran into a very bad issue over the last few days with ganesha that was working fine for the past year or so. I recently upgraded to the 2.8.3 from 2.8.2 about a month ago. During that time I did a quick audit on my configs for ganesha ceph fsal. I noticed I was missing one parameter that was in the example ceph.conf that was the Attr_Expiration_Time. I added that into the EXPORT_DEFAULTS section. Afterwards, I noticed over the past month a message about clients failing to respond to cache pressure. I did research on it and got the consensus that it wasn't an important error message. The nfs server sat relatively dorment during that month besides last week when a user ran some relatively high io traffic in the 400+ MiB/s range. 

This large amount of traffic would run fine for a few minutes then completely hang up and stop with MDS reporting slow requests upwards of 30 requests plus blocked for {time} until I would restart the mds that would hangup. I have 3 MDSes, so I would see all three sometimes with that same error message. Restarting them one at a time would clear it out then it would run for a minute or so then the slow requests would pop up.

I started looking into the mds ops and would see that the blocked requests were either RDLOCK or WRLOCK. The MDS would report that then just pause all requests. After many hours trying to figure this out, I dropped to 1 MDS after evicting all clients and ran scrub on the filesystem and found a massive amount of corrupt metadata with problem backtrace. Important to say the ceph cluster was healthy during this entire time. Also, all of these error messages with client pressure and everything else were coming from the ganesha client and not any kernel mounts. 

I repaired the metadata using scrub and went back to 3 MDSes. Started everything back up and had the user run the job again. A minute or so later, the Slow requests would pop up again. I dropped to 1 MDS again just to see if there was corrupt metadata again. This time it was only 1000 files compared to 20000 before with problem 'Backtrace'. I decided to stay with 1 MDS to see if it would fix it. The job ran perfectly fine but the failing to respond to cache pressure message still existed. I finally remembered I had set the Attr_Expiration_Time to 0 and decided to remove it. I had the user run the job again and didn't see the cache pressure message nor any slow requests and the job completed successfully. I decided to move to 3 MDSes again ran the job and neither slow requests or cache pressure issues showed up at all.

Something related to the attribute time parameter and multiple mdses is causing blocked requests due to WRLOCK and RDLOCK. Also failing to respond to client pressure messages were apparent the entire time with that parameter set. The metadata damage as a result of this was 'Backtrace'.

Ganesha config:
```
LOG 
{
	Default_Log_Level = DEBUG;
}
NFS_CORE_PARAM
{
	Enable_NLM = false;
	Enable_RQUOTA = false;
	Protocols = 4;
}
NFSv4
{
	Delegations = true;
	RecoveryBackend = rados_ng; 
	Minor_Versions =  1,2;
}
MDCACHE {
	Dir_Chunk = 0;
	NParts = 1;
	Cache_Size = 1;
}
RADOS_URLS
{
	Ceph_Conf = /etc/ceph/ceph.conf;
	UserId = "admin";
	watch_url = "rados://nfs-ganesha/ganesha-export-counter";
}
EXPORT_DEFAULTS
{
         Attr_Expiration_Time = 0;
}
%url rados://nfs-ganesha/ganesha-export-index
CEPH
{
}
RADOS_KV
{
	Ceph_Conf = /etc/ceph/ceph.conf;
	UserId = "admin";
	pool = "nfs-ganesha";
}
NFS_KRB5
{
       Active_krb5 = false;
}

Export rados example.
EXPORT {
    CLIENT {
        Clients = {ips}
        Access_Type = "rw";
    }

    SecType = "sys";
    Pseudo = "/volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7";
    Squash = "None";
    FSAL {
        Secret_Access_Key = {secret}
        User_Id = {user}
        Name = "Ceph";
    }
    Tag = "share-a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7";
    Path = "/volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7";
    Export_Id = 1022;
}
```

LOGS:
Ceph only reported the rdlock/wrlock mds dump ops, Failing to respond to cache pressure, and slow requests reported. All of which coming from ganesha clients

Ganesha logs were normal and nothing out of the ordinary except:
```
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_594] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_591] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_593] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_590] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_599] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:41 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_572] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
03/04/2020 13:57:51 : epoch 5e866ee0 : jy-cmp04.isg.siue.edu : ganesha.nfsd-1096024[svc_595] posix2fsal_error :FSAL :CRIT :Mapping 108(default) to ERR_FSAL_SERVERFAULT
```

**Comment by dang** [comment]
This is kind of odd, as the only result of setting that parameter is that Ganesha will issue many more getattr calls to libcephfs, because we're not caching attributes.  However, my understanding is that libcephfs has it's own cache, so this shouldn't affect the traffic to the MDS at all.  @jtlayton any ideas here?

**Comment by jtlayton** [comment]
In most usage patterns, libcephfs would keep a cache and not thrash the MDS too hard, but if you had a lot of competing accesses by different ceph clients then you could end up with it thrashing caps between clients, constantly granting and revoking them almost immediately. That doesn't really sound like the problem here though.

In any case, I don't have any immediate ideas, but I've had a couple of reports like this recently. I have to wonder if this is just a bug in libcephfs. v14.2.8 is pretty recent though, so it's not likely to be something known at this point. What we need at this point is a way to reliably reproduce the problem so we can understand it.

@lebonez : you said "Something related to the attribute time parameter and multiple mdses is causing blocked requests due to WRLOCK and RDLOCK. Also failing to respond to client pressure messages were apparent the entire time with that parameter set. The metadata damage as a result of this was 'Backtrace'."

Can you paste the actual messages in here (or attach the log where it said it)? We need a better idea of what the MDS is asking the client to do.

One possibility I guess is that by turning off attr caching, ganesha is hitting the libcephfs caches much more frequently, and that may be rotating dentries on the LRU such that they aren't expiring properly, but that's total speculation at this point.

**Comment by lebonez** [comment]
Here is the logs of one of the mdses when this happens. After that the only thing that helps is restarting each of the mdses. I am trying to find where I had debug on.

```
`2020-04-03 12:54:03.800 7f924e082700  1 mds.my_mds Updating MDS map to version 26385 from mon.2
2020-04-03 12:54:55.892 7f924e082700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x100009d20bc /volumes/_nogroup/9209657c-3804-4c42-99e3-d669f936a3f4/home/dchace/ [2,head] auth{0=1,1=1} v=359217554 cv=359217554/359217554 dir_auth=2 state=1074003969|complete f(v0 m2020-04-01 13:55:52.697184 19=6+13) n(v40 rc2020-04-01 14:00:28.915963 b10081453 178=96+82) hs=19+0,ss=0+0 | child=1 frozen=0 subtree=1 importing=0 replicated=1 waiter=0 authpin=0 tempexporting=0 0x556dc0d0cd00]
2020-04-03 12:54:55.892 7f924e082700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x2000010f044 /volumes/_nogroup/9209657c-3804-4c42-99e3-d669f936a3f4/home/mterry/.cache/ [2,head] auth{0=1} v=239 cv=239/239 dir_auth=2 state=1074003969|complete f(v0 m2020-04-01 16:37:39.203702 3=0+3) n(v16 rc2020-04-01 16:38:36.708976 b29693106 513=498+15) hs=3+0,ss=0+0 | child=1 frozen=0 subtree=1 importing=0 replicated=1 0x556dceeda800]
2020-04-03 12:54:59.712 7f924e082700  1 mds.my_mds Updating MDS map to version 26386 from mon.2
2020-04-03 12:54:59.744 7f9247875700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x100009d21b0 /volumes/_nogroup/9209657c-3804-4c42-99e3-d669f936a3f4/home/xyan/ [2,head] auth{0=1} v=60 cv=0/0 dir_auth=2 state=1074003969|complete f(v0 m2020-03-16 15:34:07.480659 5=4+1) n(v2 rc2020-03-16 15:34:07.482124 b2539 7=6+1) hs=5+0,ss=0+0 | child=1 frozen=0 subtree=1 importing=0 replicated=1 0x556dde522f00]
2020-04-03 12:54:59.749 7f924e082700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x10000a47b67 /volumes/_nogroup/9209657c-3804-4c42-99e3-d669f936a3f4/home/mdorris/ [2,head] auth{0=1,1=1} v=359198520 cv=0/0 dir_auth=2 state=1074003969|complete f(v22 m2020-03-31 10:53:47.971836 26=5+21) n(v53 rc2020-03-31 11:53:25.016615 b93992832 566=466+100) hs=26+0,ss=0+0 | child=1 frozen=0 subtree=1 importing=0 replicated=1 waiter=0 authpin=0 tempexporting=0 0x556de45ad900]
2020-04-03 12:54:59.762 7f9247875700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x100009d20b0 /volumes/_nogroup/9209657c-3804-4c42-99e3-d669f936a3f4/home/kcole/ [2,head] auth{0=1,1=1} v=359216657 cv=0/0 dir_auth=2 state=1073741825|complete f(v0 m2020-03-27 08:42:44.280677 31=8+23) n(v69 rc2020-03-27 09:43:00.819324 b29530096 432=284+148) hs=31+0,ss=0+0 | child=1 frozen=0 subtree=1 importing=0 replicated=1 waiter=0 authpin=0 tempexporting=0 0x556de45ade00]
2020-04-03 12:54:59.780 7f924e082700  0 mds.2.migrator nicely exporting to mds.1 [dir 0x20000111e7c /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput_v2/p450/p450_periphery_diltiazem_C5/ [2,head] auth{1=1} v=334 cv=0/0 dir_auth=2 state=1610874881|complete f(v0 m2020-04-03 12:53:40.921225 19=18+1) n(v5 rc2020-04-03 12:54:48.031564 b13453219652 59=58+1) hs=19+0,ss=0+0 dirty=19 | child=1 frozen=0 subtree=1 importing=0 replicated=1 dirty=1 authpin=0 0x556dde26a000]
2020-04-03 12:54:59.791 7f924e082700  0 mds.2.migrator nicely exporting to mds.1 [dir 0x3000007974c /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput_v2/p450/p450_periphery_diltiazem_C11/ [2,head] auth{1=1} v=1070 cv=0/0 dir_auth=2 state=1610874881|complete f(v0 m2020-04-03 12:36:36.004132 89=88+1) n(v3 rc2020-04-03 12:36:36.238133 b32896669304 129=128+1) hs=89+0,ss=0+0 dirty=89 | child=1 frozen=0 subtree=1 importing=0 exportbound=0 replicated=1 dirty=1 waiter=0 authpin=0 tempexporting=0 0x556de4767200]
2020-04-03 12:55:04.948 7f924e082700  1 mds.my_mds Updating MDS map to version 26387 from mon.2
2020-04-03 12:55:04.962 7f9247875700  0 mds.2.migrator nicely exporting to mds.1 [dir 0x20000111e79 /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput_v2/p450/rmsd/ [2,head] auth{1=1} v=10 cv=0/0 dir_auth=2 state=1610874881|complete f(v0 m2020-04-03 12:53:39.153216 2=2+0) n(v1 rc2020-04-03 12:53:39.172216 b1663 2=2+0) hs=2+0,ss=0+0 dirty=2 | child=1 frozen=0 subtree=1 importing=0 replicated=1 dirty=1 authpin=0 0x556df705ad00]
2020-04-03 12:55:04.962 7f9247875700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x10000003c9f.011* /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput/analysis_apo_bad/ [2,head] auth{0=1} v=100357 cv=100357/100357 state=1074003969|complete f(v564 m2020-04-03 10:25:01.563336 141=51+90) n(v1169 rc2020-04-03 10:25:03.285532 b5894969915 720=630+90) hs=141+0,ss=0+0 | child=1 frozen=0 subtree=0 importing=0 importbound=0 sticky=0 replicated=1 0x556db5a20000]
2020-04-03 12:55:04.972 7f9247875700  0 mds.2.migrator nicely exporting to mds.0 [dir 0x10000003c9f.010* /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput/analysis_apo_bad/ [2,head] auth{0=1} v=100053 cv=100053/100053 state=1074003969|complete f(v564 m2020-04-02 20:19:53.757356 108=44+64) n(v1169 rc2020-04-02 20:19:53.757356 b4149329295 562=494+68) hs=108+0,ss=0+0 | child=1 frozen=0 subtree=0 importing=0 importbound=0 sticky=0 replicated=1 0x556ddfbe1700]
2020-04-03 12:55:04.973 7f9247875700  0 mds.2.migrator nicely exporting to mds.1 [dir 0x10000003c9f.001* /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput/analysis_apo_bad/ [2,head] auth{0=1,1=1} v=100004 cv=100004/100004 state=1074003969|complete f(v564 m2020-04-02 20:19:53.828356 110=50+60) n(v1169 rc2020-04-02 20:19:53.828356 b3992791389 501=440+61) hs=110+0,ss=0+0 | dnwaiter=0 child=1 frozen=0 subtree=0 importing=0 importbound=0 exportbound=0 sticky=0 subtreetemp=0 replicated=1 dirty=0 waiter=0 authpin=0 tempexporting=0 0x556db6517e00]
2020-04-03 12:55:04.974 7f9247875700  0 mds.2.migrator nicely exporting to mds.1 [dir 0x10000003c9f.111* /volumes/_nogroup/a6c48cd4-17e8-49df-9a4a-0f8f8b4364c7/namdoutput/analysis_apo_bad/ [2,head] auth{0=1,1=1} v=100256 cv=100250/100250 state=1610874881|complete f(v564 m2020-04-03 12:33:57.810535 115=49+66) n(v1169 rc2020-04-03 12:33:58.402663 b563047554351 711=630+81) hs=115+0,ss=0+0 dirty=1 | dnwaiter=0 child=1 frozen=0 subtree=0 importing=0 exportbound=0 sticky=0 subtreetemp=0 replicated=1 dirty=1 waiter=0 authpin=0 tempexporting=0 0x556db4e9a500]
2020-04-03 12:55:26.892 7f924e082700  1 mds.my_mds Updating MDS map to version 26388 from mon.2
2020-04-03 12:56:04.607 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 1 included below; oldest blocked for > 33.710154 secs
2020-04-03 12:56:04.607 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:04.607 7f924b87d700  0 log_channel(cluster) log [WRN] : slow request 33.710154 seconds old, received at 2020-04-03 12:55:30.899153: client_request(client.106076219:57646 rename #0x30000079b14/p450_apoN8_lag10_spearmancorr.npy #0x10000003c9f/p450_apoN8_lag10_spearmancorr.npy 2020-04-03 12:55:30.898953 caller_uid=1002, caller_gid=1000{4,24,27,30,44,46,108,124,1000,1006,}) currently waiting for remote wrlocks
2020-04-03 12:56:04.607 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:06.807 7f924e082700  1 mds.my_mds Updating MDS map to version 26389 from mon.2
2020-04-03 12:56:09.607 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 38.710279 secs
2020-04-03 12:56:09.607 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:14.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 43.710393 secs
2020-04-03 12:56:14.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:19.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 48.710497 secs
2020-04-03 12:56:19.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:23.678 7f924e082700  1 mds.my_mds Updating MDS map to version 26390 from mon.2
2020-04-03 12:56:24.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 53.710622 secs
2020-04-03 12:56:24.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:29.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 58.710770 secs
2020-04-03 12:56:29.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:34.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 1 included below; oldest blocked for > 63.710857 secs
2020-04-03 12:56:34.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:34.609 7f924b87d700  0 log_channel(cluster) log [WRN] : slow request 63.710856 seconds old, received at 2020-04-03 12:55:30.899153: client_request(client.106076219:57646 rename #0x30000079b14/p450_apoN8_lag10_spearmancorr.npy #0x10000003c9f/p450_apoN8_lag10_spearmancorr.npy 2020-04-03 12:55:30.898953 caller_uid=1002, caller_gid=1000{4,24,27,30,44,46,108,124,1000,1006,}) currently waiting for remote wrlocks
2020-04-03 12:56:34.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:39.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 68.710971 secs
2020-04-03 12:56:39.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:44.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 1 slow requests, 0 included below; oldest blocked for > 73.711083 secs
2020-04-03 12:56:44.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 3 slow requests, 2 included below; oldest blocked for > 78.711191 secs
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) log [WRN] : slow request 33.714826 seconds old, received at 2020-04-03 12:56:15.895517: rejoin:mds.1:60963 currently dispatched
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) log [WRN] : slow request 33.714731 seconds old, received at 2020-04-03 12:56:15.895612: rejoin:mds.1:60964 currently dispatched
2020-04-03 12:56:49.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:50.042 7f924e082700  1 mds.my_mds Updating MDS map to version 26391 from mon.2
2020-04-03 12:56:51.057 7f924e082700  1 mds.my_mds Updating MDS map to version 26392 from mon.2
2020-04-03 12:56:54.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 3 slow requests, 0 included below; oldest blocked for > 83.711308 secs
2020-04-03 12:56:54.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:56:59.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 3 slow requests, 0 included below; oldest blocked for > 88.711419 secs
2020-04-03 12:56:59.609 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:57:04.610 7f924b87d700  0 log_channel(cluster) log [WRN] : 3 slow requests, 0 included below; oldest blocked for > 93.711534 secs
2020-04-03 12:57:04.610 7f924b87d700  0 log_channel(cluster) do_log log to syslog
2020-04-03 12:57:09.609 7f924b87d700  0 log_channel(cluster) log [WRN] : 3 slow requests, 0 included below; oldest blocked for > 98.711660 secs`
```

**Comment by jtlayton** [comment]
Ok, so this MDS is waiting to do a rename, and it needs a wrlock that's being held by a remote MDS. That MDS is not granting it for some reason. We may need to see the logs from the same time period on the other MDS's (the auth MDS in particular, if you know which one is).

This is most likely not a problem with ganesha at all, but the attr_expiration_time setting may be helping to expose this somehow. What may be best is open a new bug for this at https://tracker.ceph.com. It'll be easier to get the right eyes on this there, and we'll need that anyway if the problem is in ceph and not ganesha.

**Comment by lebonez** [comment]
will do thanks!

**Comment by jtlayton** [comment]
@lebonez : when you open the tracker bug, can you paste a link to it in here? Others may be interested in following along.

**Comment by lebonez** [comment]
Sure I am working on writing it up now my account was activated an hour ago. I do have an update. I am actually still seeing the failing to respond to cache pressure error even after moving to 1 mds and the ATTR off. I also did another scrub on the filesystem and found the backtrace problem metadata. I do not see the slow requests or block WRLOCK/RDLOCK with 1 MDS. As soon as I saw the cache pressure, I moved back to 1 MDS before the jobs were run by the user again.
